pkg/api/validation/validation.go

```diff
@@ -2844,10 +2799,16 @@ func ValidateNodeUpdate(node, oldNode *api.Node) field.ErrorList {
 	// Clear status
 	oldNode.Status = node.Status
 
+	// update taints
```

Comment:
mutable pod specs is a pretty fundamental shift. I would maintain the status quo for this PR and propose allowing mutation separately.